vllm/v1/sample/sampler.py

```diff
@@ -117,11 +117,17 @@ def sample(
         return sampled
 
     def compute_probs(self, logits: torch.Tensor,
-                      sampling_metadata: SamplingMetadata) -> torch.Tensor:
+                      sampling_metadata: SamplingMetadata,
+                      sample_lens: list[int]) -> torch.Tensor:
         if sampling_metadata.all_greedy:
             return logits
-        # Apply temperature. This is an in-place op changing logits.
-        logits = self.apply_temperature(logits, sampling_metadata.temperature)
+        assert sampling_metadata.temperature is not None
+        temperature = torch.repeat_interleave(
+            sampling_metadata.temperature,
+            torch.tensor(sample_lens,
+                         device=sampling_metadata.temperature.device))
```

Comment:
This will cause CPU -> GPU synchronization. While we don't have to optimize it in this PR, I think it's worth a comment.